### ISSUE TITLE
Enforcing Small batch size, issue with batch size getting reset to number of users

### ISSUE THREAD

**Issue by shahjaidev** [THUwangcy/DirectAU#5]
(Moving this to a separate issue for better relevance :) )

I see the following warning that Batch size is changed to 2635632 (number of users). Is this by design? I want to use a much smaller batch size so training is faster, as currently it take a very long time per epoch. (As torch.pdist in the uniformity loss is O(B^2) where B is the batch size)

How would you suggest enforcing this?  (Currently, setting a batch size either as a command arg or in the config gets overwritten it seems)

![image](https://user-images.githubusercontent.com/22052444/185252934-ec08b9ad-c011-4ea9-a459-34930d4717a6.png)

Where in the code is this batch size re-set?  (_batch_size_adaptation ?)
Are there any changes to the dataloader in this repository to the standard RecBole general dataloder

**Comment by THUwangcy** [comment]
From my understanding of RecBole, the batch size here means eval_batch_size. If possible, it will set the eval_batch_size to the whole item set to accelerate the non-sampling evaluation. 

As for the batch size for training, you can set `--train_batch_size=xxx` from the command line, and check whether the output log (Training Hyper Parameters -> train_batch_size) is as expected. The training progress also shows the total number of batch on the right. When I run with different train_batch_size, the number of batch changes as expected.
